### PR TITLE
Connection tests

### PR DESCRIPTION
**Pull request by freshfab**
Adds tests for the connection package.

### CODE REVIEW

File: privval/sc_file.go [hunk 1]
```diff
@@ -111,6 +111,7 @@ func (pv *SCFilePV) run() {
 
 			var err error
 			if pv.SecretConn, err = connection.RetryDial(
+				config.Dir(),
```
Comment (dee-effes) [inline privval/sc_file.go:114]: Is there a reason for this line?  the returning string is not used

File: privval/sc_file.go [hunk 2]
```diff
@@ -165,6 +166,7 @@ func (pv *SCFilePV) OnStart() (err error) {
 
 	// Dial the validator.
 	if pv.SecretConn, err = connection.RetryDial(
+		config.Dir(),
```
Comment (dee-effes) [inline privval/sc_file.go:169]: same as above

File: privval/sc_file.go (hunk 1 above)
Comment (freshfab) [reply privval/sc_file.go:114]: Not sure why you think it's not used. The config directory passed into RetryDial is used to locate the conn.key file when using a TCP connection between SignCTRL and the validator.

Comment (dee-effes) [reply privval/sc_file.go:114]: Ah, I didn't see that it's an argument. Thought we were in the if block, sorry.

**Pull request merged**